include/caffe/layers/cudnn_conv_layer.hpp

```diff
@@ -86,6 +88,7 @@ class CuDNNConvolutionLayer : public ConvolutionLayer<Dtype> {
   size_t workspaceSizeInBytes;  // size of underlying storage
   void *workspaceData;  // underlying storage
   void **workspace;  // aliases into workspaceData
+  miopenHandle_t handle_;
```

Comment:
- remove `stream_` so we only use default HIP stream.
- change `handle_` from an array to one single instance so fwd / bwd weight / bwd data in a single layer share the same MIOpen handle.